File: AtCoder/Greens/cpp/ABC035-B-Drone.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

string S;
ll T;

int main() {
  cin >> S >> T;
  ll h = 0, w = 0, q = 0;
  for (auto s : S) {
    if (s == 'U')
      ++h;
    else if (s == 'D')
      --h;
    else if (s == 'R')
      ++w;
    else if (s == 'L')
      --w;
    else
      ++q;
  }
  ll res = 0;
  if (T == 2) {
    res = abs(h) + abs(w) - q;
    if (res < 0) res = abs(res) % 2;
  } else {
    res = abs(h) + abs(w) + q;
  }
  printf("%lld\n", res);
}

File: AtCoder/Greens/python/ABC035-B-Drone.py
S = input()
T = int(input())
h, w, q = 0, 0, 0
for s in S:
    if s == 'L':
        w -= 1
    elif s == 'R':
        w += 1
    elif s == 'U':
        h += 1
    elif s == 'D':
        h -= 1
    else:
        q += 1
if T == 1:
    res = abs(h) + abs(w) + q
else:
    res = abs(h) + abs(w) - q
    if res < 0:
        res = abs(res) % 2
print(res)
